sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/PlanService.java

```diff
@@ -1,14 +1,52 @@
 package com.kyriba.conference.sponsorship.service;
 
+import com.kyriba.conference.sponsorship.dao.PlanRepository;
+import com.kyriba.conference.sponsorship.domain.Plan;
+import com.kyriba.conference.sponsorship.domain.PlanCategory;
+import com.kyriba.conference.sponsorship.domain.Sponsor;
+import org.springframework.lang.Nullable;
 import org.springframework.stereotype.Service;
 
+import javax.transaction.Transactional;
+
 
 /**
  * @author M-ASL
  * @since v1.0
  */
 @Service
+@Transactional

```

Comment:
@xpinjection can we use class-level annotation if there is some non-transactional methods?

Reply:
There is no such thing as non-transactional methods, there are different transactional boundaries. ;)

Reply:
Could you explain what's happening with 'readSomething()'? A new transaction opens or not?

Reply:
DB transaction must always present when call is made to DB. There are 3 options:

- autoCommit after each statement (default option in JDBC);
- read only transaction started manually (with Transactional annotation);
- normal transaction started manually (with Transactional annotation);